src/transformers/modeling_utils.py

```diff
@@ -3737,6 +3749,9 @@ def _fix_key(key):
             for pat in cls._keys_to_ignore_on_load_unexpected:
                 unexpected_keys = [k for k in unexpected_keys if re.search(pat, k) is None]
 
+        if hf_quantizer is not None:
+            missing_keys = hf_quantizer.update_missing_keys(model, missing_keys, prefix)
```

Comment:
Not sure this is something we want / I understand why this is needed

Reply:
I can try to sneak that in inside the `hf_quantizer.create_quantized_param` just like how bnb deal with `unexpected_keys`.

Reply:
The buffer are created at the moment we replace the layers so modifying the missing keys here `hf_quantizer.create_quantized_param` don't make sense + it does not cover some edges cases . If possible, I prefer leaving it like this. LMK if this is possible or not.

Reply:
Alright no worries 🤗